test/extensions/transport_sockets/tls/ssl_socket_test.cc

```diff
@@ -3453,14 +3453,182 @@ TEST_P(SslSocketTest, ClientAuthMultipleCAs) {
 
 namespace {
 
+/**
+ * Verifies additional SSL connection info expectations.
+ */
+class SslExpectations {
+public:
+  SslExpectations& setExpectedPeerCertificateValidated(bool expected_peer_certificate_validated) {
+    expected_peer_certificate_validated_ = expected_peer_certificate_validated;
+    return *this;
+  }
+
+  void verify(const Ssl::ConnectionInfoConstSharedPtr ssl) const {
+    if (expected_peer_certificate_validated_.has_value()) {
+      EXPECT_EQ(expected_peer_certificate_validated_.value(), ssl->peerCertificateValidated());
+    }
+  }
+
+private:
+  absl::optional<bool> expected_peer_certificate_validated_{};
+};
+
+static bssl::UniquePtr<SSL_SESSION> last_ssl_session;
```

Comment:
Storing as a static is problematic; make it a member of the test class. You'll need a little bit more code to store some data in the SSL in a new index probably, but it will be much cleaner overall.